src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,2091 @@
+
+        # apply default scale where there are no observations
+        scale = torch.where(num_observed > 0, scale, default_scale)
+
+        # ensure the scale is at least `self.minimum_scale`
+        scale = torch.clamp(scale, min=self.minimum_scale)
+        scaled_data = data / scale
+
+        if not self.keepdim:
+            scale = scale.squeeze(dim=self.dim)
+
+        return scaled_data, torch.zeros_like(scale), scale
+
+
+# Copied from transformers.models.time_series_transformer.modeling_time_series_transformer.TimeSeriesNOPScaler with TimeSeries->PatchTST
+class PatchTSTNOPScaler(nn.Module):
+    """
+    Assigns a scaling factor equal to 1 along dimension `dim`, and therefore applies no scaling to the input data.
+
+    Args:
+        dim (`int`):
+            Dimension along which to compute the scale.
+        keepdim (`bool`, *optional*, defaults to `False`):
+            Controls whether to retain dimension `dim` (of length 1) in the scale tensor, or suppress it.
+    """
+
+    def __init__(self, dim: int, keepdim: bool = False):
+        super().__init__()
+        self.dim = dim
+        self.keepdim = keepdim
+
+    def forward(
+        self, data: torch.Tensor, observed_indicator: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        scale = torch.ones_like(data, requires_grad=False).mean(dim=self.dim, keepdim=self.keepdim)
+        loc = torch.zeros_like(data, requires_grad=False).mean(dim=self.dim, keepdim=self.keepdim)
+        return data, loc, scale
+
+
+@add_start_docstrings(
+    "The bare PatchTST Model outputting raw hidden-states without any specific head.",
+    PATCHTST_START_DOCSTRING,
+)
+class PatchTSTModel(PatchTSTPreTrainedModel):
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__(config)
+
+        if config.scaling == "mean" or config.scaling is True:
+            self.scaler = PatchTSTMeanScaler(dim=1, keepdim=True)
+        elif config.scaling == "std":
+            self.scaler = PatchTSTStdScaler(dim=1, keepdim=True)
+        else:
+            self.scaler = PatchTSTNOPScaler(dim=1, keepdim=True)
+
+        self.patching = PatchTSTPatchify(
+            config.context_length,
+            patch_length=config.patch_length,
+            stride=config.stride,
+        )
+        self.mask_input = config.mask_input
+
+        if self.mask_input:
+            self.masking = PatchTSTMasking(
+                mask_type=config.mask_type,
+                mask_ratio=config.mask_ratio,
+                mask_patches=config.mask_patches,
+                mask_patch_ratios=config.mask_patch_ratios,
+                channel_consistent_masking=config.channel_consistent_masking,
+                unmasked_channel_indices=config.unmasked_channel_indices,
+                mask_value=config.mask_value,
+                seed_number=config.seed_number,
+            )
+        else:
+            self.masking = nn.Identity()
+        self.encoder = PatchTSTEncoder(config)
```

Comment:
```suggestion
        self.embeddings = PatchTSTEmbeddings(config)
        self.encoder = PatchTSTEncoder(config)
```
If we define it in this way, then it's equivalent to how BERT is implemented

Reply:
Thank you for the suggestion, it is resolved.